### ISSUE TITLE
Action Event Pruning causes timeout on bulk action runs

### ISSUE THREAD

**Issue by ragingdave** [laravel/nova-issues#1674]
- Laravel Version: 5.8.23
- Nova Version: 2.0.6
- PHP Version: 7.2
- Operating System and Version: CentOS 7
- Browser type and version: Chrome

### Description:
The pruning that occurs for action events is generating hundreds of queries on an action running that can cause the UI to hang as a result.
This occurs on any action event run and as it runs per item selected for the action, it ends up 
running as many individual queries to trim the action event log as models selected. While each prune query only takes a few milliseconds, this adds up to (via telescope and set_time_limit extension) almost 1 minute for ~700 models selected for an action. Note that this action implements ShouldQueue, and in this test, I didn't even trigger my application logic in the handle method at all, it's just bare Nova specific pruning logic.

### Steps To Reproduce:
Have a model with about 1000 models, try and run an action on all of them, note that under default php settings, the UI hangs and waits for >1000 queries to be run for pruning and potentially get a "maximum execution time exceeded" error.

Perhaps if the pruning happened, either in bulk (somehow, not sure if possible based on the way it queries), or at least in the action at handle time, this could be alleviated. Or even if the pruning happened during a batch being marked as failed/completed somehow, but either way on large selection of models, actions break down

**Comment by jbrooksuk** [comment]
You can control the frequency of pruning by setting the static `chunkCount` on your custom action class.

The higher the number, the less frequently pruning will occur.

**Comment by ragingdave** [comment]
In revisiting this issue from my own code comments, I don't think this resolves the issue. The issue isn't the number of items to chunk at the action level, it has to do with the number of prune calls are handled. Each model in the chunk will result in a query to delete older action events for each. This in turn causes way more queries than a single request might be able to handle. It would likely be better to handle that as a queue job to prune a specific model in like a chain, or dispatching a batch set to make it so the user isn't waiting. 

Right now, basically you are forced to use a queued action to even hope to prevent this error on larger selections, but that has it's own caveats with not being able to accept a file upload, so it's kind of a catch 22.

Basically it would be nice if this got a second look for orion or at the very least a rework in the next release set.
